File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_Treasury.java
package com.broll.gainea.server.core.cards.impl.play;

import com.broll.gainea.server.core.cards.Card;

public class C_Treasury extends Card {


    public C_Treasury() {
        super(3, "Reichtum", "Erhaltet ein weiteres Ziel");
        setDrawChance(0.5f);
    }

    @Override
    public boolean isPlayable() {
        return true;
    }

    @Override
    protected void play() {
       game.getGoalStorage().assignNewRandomGoal(owner);
    }

}

File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_Treasury.kt
package com.broll.gainea.server.core.cards.impl.play

import com.broll.gainea.server.core.cards.Card
import com.broll.gainea.server.core.cards.EffectType.OTHER

class C_Treasury : Card(3, OTHER, "Reichtum", "Erhaltet ein weiteres Ziel") {
    init {
        drawChance = 0.5f
    }

    override val isPlayable: Boolean
        get() = true

    override fun play() {
        game.goalStorage.assignNewRandomGoal(owner)
    }
}
